client/rpc_sender.go

```diff
@@ -83,6 +84,13 @@ func newRPCSender(server string, context *base.Context, retryOpts retry.Options,
 // and been executed successfully. We retry here to eventually get through with
 // the same client command ID and be given the cached response.
 func (s *rpcSender) Send(ctx context.Context, ba roachpb.BatchRequest) (*roachpb.BatchResponse, *roachpb.Error) {
+	timeout := 5 * s.retryOpts.MaxBackoff
```

Comment:
Confirmed, when I heavily load my machine and run in race mode, the handshake takes up to 750ms while the tests require that the client become ready in less than 250ms.

All that said, I think this patch is the best we can hope for until go 1.6.